library/blocks/block-extensions/src/main/java/org/quiltmc/qsl/block/extensions/api/QuiltBlockSettings.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.block.extensions.api;
+
+import net.minecraft.block.*;
+import net.minecraft.entity.EntityType;
+import net.minecraft.sound.BlockSoundGroup;
+import net.minecraft.util.DyeColor;
+import net.minecraft.util.Identifier;
+import org.quiltmc.qsl.block.extensions.mixin.AbstractBlockAccessor;
+import org.quiltmc.qsl.block.extensions.mixin.AbstractBlockSettingsAccessor;
+
+import java.util.function.Function;
+import java.util.function.ToIntFunction;
+
+public class QuiltBlockSettings extends AbstractBlock.Settings {
+	protected QuiltBlockSettings(Material material, Function<BlockState, MapColor> mapColorProvider) {
+		super(material, mapColorProvider);
+	}
+
+	protected QuiltBlockSettings(Material material, MapColor mapColor) {
+		super(material, mapColor);
+	}
+
+	protected QuiltBlockSettings(AbstractBlock.Settings settings) {
+		super(((AbstractBlockSettingsAccessor) settings).getMaterial(), ((AbstractBlockSettingsAccessor) settings).getMapColorProvider());
+
+		// [VanillaCopy] AbstractBlock.Settings#copy(AbstractBlock.Settings)
+		AbstractBlockSettingsAccessor thisAccessor = (AbstractBlockSettingsAccessor) this;
```

Comment:
I would use `var` here to shorten the lines and the cast make it obvious which type the local is.
```suggestion
		var thisAccessor = (AbstractBlockSettingsAccessor) this;
```